### PR TITLE
fix partition table with partition column name date

### PR DESCRIPTION
**Pull request by marsishandsome**
### What problem does this PR solve? 
fix https://asktug.com/t/topic/67814/6

### What is changed and how it works?
partition column name should wrapped with `

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

### CODE REVIEW

**Review by birdstorm — DISMISSED** [review]
LGTM

File: tikv-client/src/main/java/com/pingcap/tikv/expression/PartitionPruner.java [hunk 1]
```diff
@@ -80,15 +80,27 @@ static void generateRangeExprs(
...
+            String.format("%s >= %s and %s", wrapColumnName(partExprStr), previous, leftHand);
         partExprs.add(parser.parseExpression(and));
       }
     }
   }
+
+  private static String wrapColumnName(String columnName) {
+    if (columnName.startsWith("`") && columnName.endsWith("`")) {
+      return columnName;
+    } else if (columnName.contains("(") && columnName.contains(")")) {
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/expression/PartitionPruner.java:99]: what if columnName = `a + b`?

Comment (marsishandsome) [reply tikv-client/src/main/java/com/pingcap/tikv/expression/PartitionPruner.java:99]: https://docs.pingcap.com/tidb/stable/partitioned-table#partitioning-limitations-relating-to-functions

image

`+` is not a `partition function` .

**Pull request merged**